azurerm/internal/services/hdinsight/tests/hdinsight_hadoop_cluster_resource_test.go

```diff
@@ -1163,6 +1219,8 @@ func testAccAzureRMHDInsightHadoopCluster_template(data acceptance.TestData) str
 	return fmt.Sprintf(`
 provider "azurerm" {
   features {}
+  # Due to bug in HDInsight we can't reuse client with the same x-ms-correlation-request-id for multiple updates
+  disable_correlation_request_id = true
```

Comment:
Is there a public GH issue tracking this for resolution that we can leave in the comments here?

Reply:
I'm not aware about GH repo with public HDI issue. I got that information from Microsoft support and then Product group when I hit the issue. It's more like limitation than a bug. AFAIK, it's by design.

Reply:
rather than doing this within the test suite, we should be able to update the [client](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/hdinsight/client/client.go#L15-L34) to not send this:

```
func NewClient(o *common.ClientOptions) *Client {
	// due to a bug in the HDInsight API we can't reuse client with the same x-ms-correlation-request-id for multiple updates
	opts := *o
	opts.DisableCorrelationRequestID = true

	ApplicationsClient := hdinsight.NewApplicationsClientWithBaseURI(opts.ResourceManagerEndpoint, opts.SubscriptionId)
	opts.ConfigureClient(&ApplicationsClient.Client, opts.ResourceManagerAuthorizer)

	ClustersClient := hdinsight.NewClustersClientWithBaseURI(opts.ResourceManagerEndpoint, opts.SubscriptionId)
	opts.ConfigureClient(&ClustersClient.Client, opts.ResourceManagerAuthorizer)

	ConfigurationsClient := hdinsight.NewConfigurationsClientWithBaseURI(opts.ResourceManagerEndpoint, opts.SubscriptionId)
	opts.ConfigureClient(&ConfigurationsClient.Client, opts.ResourceManagerAuthorizer)

	ExtensionsClient := hdinsight.NewExtensionsClientWithBaseURI(opts.ResourceManagerEndpoint, opts.SubscriptionId)
	opts.ConfigureClient(&ExtensionsClient.Client, opts.ResourceManagerAuthorizer)

	return &Client{
		ApplicationsClient:   &ApplicationsClient,
		ClustersClient:       &ClustersClient,
		ConfigurationsClient: &ConfigurationsClient,
		ExtensionsClient:     &ExtensionsClient,
	}
}
```

Reply:
Done.